tools/generator/InterfaceGen.cs

```diff
@@ -468,6 +468,10 @@ public void GenerateEventsOrPropertiesForListener (StreamWriter sw, string inden
 					Report.Warning (0, Report.WarningInterfaceGen + 1, "empty event name in {0}.{1}.", FullName, method.Name);
 					continue;
 				}
+				if (opt.GetSafeIdentifier (name) != name) {
+					Report.Warning (0, Report.WarningInterfaceGen + 4, "event name for {0}.{1} is invalid.", FullName, method.Name);
```

Comment:
Warning message should include a way to "fix" the warning, e.g. by explicitly naming it (what's the attribute?).